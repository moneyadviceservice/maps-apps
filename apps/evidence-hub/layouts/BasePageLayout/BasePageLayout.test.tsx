import { render } from '@testing-library/react';

import { BasePageLayout } from '.';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    query: {},
    asPath: '/',
  }),
}));

const siteConfig = {
  seoTitle: 'test',
  seoDescription: 'test',
  headerLogo: {
    image: {
      _path: '/test.png',
      width: 20,
      height: 20,
      mimeType: 'image/png',
    },
    altText: 'SFS Logo',
  },
  mainNavigation: [{ linkTo: '/test', text: 'test', description: null }],
  footerLinks: [
    {
      title: 'test',
      childLinks: [
        { linkTo: '/test', text: 'test', description: null },
        { linkTo: '/test2', text: 'test2', description: null },
      ],
    },
  ],
};

describe('BasePageLayout', () => {
  it('renders correctly', () => {
    const { container } = render(
      <BasePageLayout
        siteConfig={siteConfig}
        assetPath=""
        pageTitle="test"
        breadcrumbs={[]}
      >
        test content
      </BasePageLayout>,
    );
    expect(container).toMatchSnapshot();
  });
});
