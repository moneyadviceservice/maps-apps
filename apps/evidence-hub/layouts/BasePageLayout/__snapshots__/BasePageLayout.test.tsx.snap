// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasePageLayout renders correctly 1`] = `
<div>
  <div
    id="top"
  >
    <header
      data-testid="header"
    >
      <a
        class="sr-only focus:not-sr-only focus:bg-yellow-400 focus:border-b-4 focus:border-b-blue-700 focus:absolute focus:p-2 focus:rounded focus:top-4 focus:left-4 focus:z-50"
        href="#main"
      >
        Skip to content
      </a>
      <div
        class="relative z-20 flex items-center px-4 lg:hidden print:hidden"
      >
        <a
          aria-label="Evidence Hub"
          class="mx-auto min-w-[170px]"
          href="/en"
        >
          <img
            alt="Evidence Hub Logo"
            data-nimg="1"
            decoding="async"
            height="77.56"
            loading="lazy"
            src="/_next/image?url=%2Ftest.png&w=384&q=75"
            srcset="/_next/image?url=%2Ftest.png&w=256&q=75 1x, /_next/image?url=%2Ftest.png&w=384&q=75 2x"
            style="color: transparent;"
            width="170"
          />
        </a>
        <div
          class="container-auto h-[134px] p-0 m-0 flex justify-center"
        >
          <details
            class="group w-[31px] ml-auto"
          >
            <summary
              class="open h-[134px] flex items-center justify-center cursor-pointer list-none [&::-webkit-details-marker]:hidden text-center text-white text-sm"
              data-testid="toggle-nav"
              title="Open menu"
            >
              <div
                class="group-open:hidden flex flex-col h-[38px] w-[38px] text-left absolute align-middle justify-center"
              >
                <svgr-mock
                  classname="text-gray-800"
                />
              </div>
              <div
                class="hidden text-center group-open:block"
              >
                <div
                  class="flex flex-col"
                >
                  <svgr-mock
                    classname="w-[30px]"
                  />
                </div>
              </div>
            </summary>
            <div
              class="bg-white absolute z-10 top-[134px] w-full left-0"
            >
              <form
                action="https://maps.org.uk/en/search-results.html"
                class="flex lg:w-[266px] w-full h-9 border-gray-600 relative z-1 border-1 rounded pr-8"
                data-testid="form-search"
                role="search"
              >
                <input
                  aria-label="Search"
                  class="w-full p-2 rounded-l"
                  id="q"
                  name="q"
                  required=""
                  type="text"
                />
                <button
                  aria-label="Search"
                  class="p-1 bg-magenta-500 text-white rounded-r absolute right-[-1px] top-[-1px] shadow-bottom-gray max-h-9 max-w-9"
                  type="submit"
                >
                  <svgr-mock
                    aria-hidden="true"
                    classname=""
                  />
                </button>
              </form>
              <div
                class="pb-10"
              >
                <nav
                  class="mb-4"
                >
                  <ul>
                    <li>
                      <a
                        class="inline-flex items-center gap-2 focus:text-gray-800 focus:bg-yellow-400 focus:shadow-link-focus focus-within:outline-none hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none py-2 w-full relative px-4 no-underline text-lg font-bold text-magenta-500 visited:text-gray-600"
                        href="/en/test"
                      >
                        test
                        <svgr-mock
                          classname="ml-auto text-magenta-500"
                        />
                      </a>
                    </li>
                  </ul>
                </nav>
                <div
                  class="flex mx-4"
                >
                  <a
                    class="group inline-flex items-center gap-2 focus-within:outline-none hover:text-pink-400 focus:text-gray-800 focus:bg-yellow-400 focus:shadow-link-focus focus-within:outline-0 hover:underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none text-gray-600 visited:text-gray-600 p-0 underline text-sm ml-auto font-bold"
                    href="/cy"
                    hreflang="cy"
                    lang="cy"
                    rel="alternate"
                  >
                    <span
                      aria-describedby="lang-description"
                      class="inline-block relative after:content-[''] after:inline-block after:absolute after:-bottom-1 after:left-0 after:bg-white after:w-full after:h-[1px] group-hover:after:h-[2px] "
                    >
                      Cymraeg
                    </span>
                  </a>
                  <span
                    class="sr-only"
                    id="lang-description"
                  >
                    Newid yr iaith ir Gymraeg
                  </span>
                </div>
              </div>
            </div>
          </details>
        </div>
      </div>
      <div
        class="hidden lg:block"
      >
        <div
          class="relative print:hidden"
        >
          <div
            class="container-auto flex items-center justify-center lg:justify-normal max-w-[1272px]"
          >
            <a
              aria-label="Evidence Hub"
              href="/en"
            >
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height="73"
                loading="lazy"
                src="/_next/image?url=%2Ftest.png&w=384&q=75"
                srcset="/_next/image?url=%2Ftest.png&w=256&q=75 1x, /_next/image?url=%2Ftest.png&w=384&q=75 2x"
                style="color: transparent;"
                width="160"
              />
            </a>
            <div
              class="items-center hidden ml-auto lg:flex py-7"
            >
              <a
                class="group inline-flex items-center gap-2 focus-within:outline-none hover:text-pink-400 focus:text-gray-800 focus:bg-yellow-400 focus:shadow-link-focus focus-within:outline-0 hover:underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none text-gray-600 visited:text-gray-600 mr-6 p-0 underline text-sm ml-6"
                href="/cy"
                hreflang="cy"
                lang="cy"
                rel="alternate"
              >
                <span
                  aria-describedby="lang-description"
                  class="inline-block relative after:content-[''] after:inline-block after:absolute after:-bottom-1 after:left-0 after:bg-white after:w-full after:h-[1px] group-hover:after:h-[2px] "
                >
                  Cymraeg
                </span>
              </a>
              <span
                class="sr-only"
                id="lang-description"
              >
                Newid yr iaith ir Gymraeg
              </span>
              <form
                action="https://maps.org.uk/en/search-results.html"
                class="flex lg:w-[266px] w-full h-9 border-gray-600 relative z-1 border-1 rounded pr-8"
                data-testid="form-search"
                role="search"
              >
                <input
                  aria-label="Search"
                  class="w-full p-2 rounded-l"
                  id="q"
                  name="q"
                  required=""
                  type="text"
                />
                <button
                  aria-label="Search"
                  class="p-1 bg-magenta-500 text-white rounded-r absolute right-[-1px] top-[-1px] shadow-bottom-gray max-h-9 max-w-9"
                  type="submit"
                >
                  <svgr-mock
                    aria-hidden="true"
                    classname=""
                  />
                </button>
              </form>
            </div>
          </div>
        </div>
        <div
          class="hidden lg:block"
        >
          <div
            class="container-auto max-w-[1272px] mt-5 mb-2"
          >
            <nav>
              <ul
                class="flex h-[40px] items-center"
              >
                <li>
                  <a
                    class="inline-flex items-center gap-2 focus-within:outline-none no-underline visited:text-white focus:text-gray-800 focus:bg-yellow-400 focus:shadow-link-focus focus-within:outline-0 focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none text-pink-700 font-bold relative mr-8 h-[40px] text-sm hover:text-green-300 hover:no-underline"
                    href="/en/test"
                  >
                    test
                  </a>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </div>
    </header>
    <div
      class="bg-blue-700"
      data-testid="title-banner"
    >
      <div
        class="container-auto max-w-[1272px] py-4"
      >
        <h1
          class="text-4xl md:text-6xl font-bold text-white"
        >
          Evidence summaries
        </h1>
      </div>
    </div>
    <div
      class="container-auto max-w-[1272px] py-4"
    >
      <main
        id="main"
      >
        test content
      </main>
    </div>
    <footer
      class="relative z-1"
      data-testid="footer"
    >
      <div
        class="flex flex-col py-8 bg-gray-800 md:py-6"
      >
        <div
          class="container-auto max-w-[1272px]"
        >
          <nav
            aria-label="Footer navigation"
            class="grid grid-cols-1 gap-8 md:grid-cols-3"
          >
            <div
              class="flex flex-col"
            >
              <h3
                class="mb-4 text-lg font-semibold text-white"
              >
                test
              </h3>
              <ul
                class="flex flex-col pl-6 space-y-1 text-white list-disc"
              >
                <li>
                  <a
                    class="inline-flex items-center gap-2 focus:text-gray-800 focus:bg-yellow-400 focus:shadow-link-focus focus-within:outline-none focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none no-underline text-sm text-white visited:text-gray-100 hover:text-green-300 hover:underline transition-colors duration-200"
                    href="/test"
                  >
                    test
                  </a>
                </li>
                <li>
                  <a
                    class="inline-flex items-center gap-2 focus:text-gray-800 focus:bg-yellow-400 focus:shadow-link-focus focus-within:outline-none focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none no-underline text-sm text-white visited:text-gray-100 hover:text-green-300 hover:underline transition-colors duration-200"
                    href="/test2"
                  >
                    test2
                  </a>
                </li>
              </ul>
            </div>
          </nav>
          <div
            class="pt-4 pb-10 mt-6 text-center border-t border-white"
          >
            <p
              class="m-0 text-white"
              data-testid="paragraph"
            >
              Copyright
               
              2025 Money & Pensions Service, Borough Hall, Cauldwell Street, Bedford, MK42 9AB.
            </p>
            <p
              class="mb-4 text-white"
              data-testid="paragraph"
            >
              All rights reserved.
            </p>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
